tfjs-backend-webgpu/src/kernels/AvgPool.ts

```diff
@@ -45,15 +45,16 @@ export function avgPool(
   }
 
   const dimensions = [
-    convInfo.padInfo.top, convInfo.padInfo.left,      // Padding.
-    convInfo.strideHeight, convInfo.strideWidth,      // Stride.
-    convInfo.dilationHeight, convInfo.dilationWidth,  // Dilation.
-    convInfo.inHeight, convInfo.inWidth,              // Conv dims.
-    convInfo.effectiveFilterHeight,
-    convInfo.effectiveFilterWidth  // Filter dims.
+    {type: 'int32', data: [convInfo.padInfo.top, convInfo.padInfo.left]},
+    {type: 'int32', data: [convInfo.strideHeight, convInfo.strideWidth]},
+    {type: 'int32', data: [convInfo.dilationHeight, convInfo.dilationWidth]},
+    {type: 'int32', data: [convInfo.inHeight, convInfo.inWidth]}, {
+      type: 'int32',
+      data: [convInfo.effectiveFilterHeight, convInfo.effectiveFilterWidth]
+    }
   ];
-  const uniformData = new Int32Array(dimensions);
-  return backend.runWebGPUProgram(program, [x], x.dtype, uniformData);
+
```

Comment:
Can it be written like below? Similar to other places.
```
  const dimensions = [
    convInfo.padInfo.top, convInfo.padInfo.left,      // Padding.
    convInfo.strideHeight, convInfo.strideWidth,      // Stride.
    convInfo.dilationHeight, convInfo.dilationWidth,  // Dilation.
    convInfo.inHeight, convInfo.inWidth,              // Conv dims.
    convInfo.effectiveFilterHeight,
    convInfo.effectiveFilterWidth  // Filter dims.
  ];
  const uniformData = {type: 'int32', data: dimensions}

Reply:
This is grouped as shader type: float/int, vec2/ivec2, vec3/ivec3, vec4/ivec4

Reply:
I am still confused that why we can't do like that. Will take with you offline.

Reply:
Got it. Thanks.